snmp/tests/test_e2e_core_profiles/utils.py

```diff
@@ -0,0 +1,25 @@
+# (C) Datadog, Inc. 2023-present
+# All rights reserved
+# Licensed under a 3-clause BSD style license (see LICENSE)
+
+from .. import common
+
+
+def create_profile_test_config(profile_name):
+    config = common.generate_container_instance_config([])
+    config['init_config']['loader'] = 'core'
+    instance = config['instances'][0]
+    instance.update({'community_string': profile_name})
```

Comment:
@NouhaManai96 

> the community_string must match the name of the .snmprec

Indeed, that's the mechanism used by snmpsim, by using a specific community_string, snmpsim will load the corresponding .snmprec.

> but does it always/necessarily match the name of the profile ?

No, it's doesn't need to match, but since we are usually testing on per profile basis, for consistency and readability, it's likely better to have them match.

Reply:
@NouhaManai96 I made some changes here https://github.com/DataDog/integrations-core/pull/14693/commits/592477824809ec8371b5560edaad86f152671689 to make it more explicite.

Let me know if that helps.

I'm not sure we should enforce the community_string to match the profile name since it might happen that we want to test 2 different scenarios (2 different .snmprec) for a same profile.